forest/cli/src/cli/wallet_cmd.rs

```diff
@@ -241,3 +278,156 @@ impl WalletCommands {
         }
     }
 }
+
+fn format_balance_string(
+    token_amount: TokenAmount,
+    mode: FormattingMode,
+) -> anyhow::Result<String> {
+    let units = ["atto", "femto", "pico", "nano", "micro", "milli", ""];
+    let num = token_amount.atto().to_string().parse::<Decimal>()?;
```

Comment:
You can convert a `TokenAmount` to a `Decimal` by going through an `i128` instead of a `String`.

Reply:
Done